pages/common/ack.md

```diff
@@ -1,32 +1,37 @@
 # ack
 
-> A search tool like grep, optimized for programmers.
-> More information: <https://beyondgrep.com/documentation/>.
+> A search tool like grep, optimized for developers.
+> See also: `rg`, which is much faster.
+> More information: <https://beyondgrep.com/documentation>.
 
-- Find files containing "foo":
+- Search for files containing a string or regular expression in the current directory and subdirectories:
 
-`ack {{foo}}`
+`ack "{{search_pattern}}"`
 
-- Find files of a specific type:
+- Search for a case-insensitive pattern:
 
-`ack --ruby {{foo}}`
+`ack --ignore-case "{{search_pattern}}"`
 
-- Count the total number of matches for the term "foo":
+- Search for lines matching a pattern, printing [o]nly the matched text and not the rest of the line:
```

Comment:
As with `grep`, I'm not convinced of the usefulness of `-o` here. Could you give some example scenarios in which it might be useful?

Reply:
I don't use `ack` (I like `rg` better), but with `grep` it's useful if you know regular expressions but you don't know the syntax of `awk`, `sed`, etc., and you just want to extract data from a file or HTTP response real quick.  Also, you can see it's used (obviously frequently) [here](https://github.com/tldr-pages/tldr/pull/5392/files#diff-6b7ae119ed0f8d762365985f1cbb0b88af178ceb5f960cce0d5e9579a2102226R4).

Reply:
Also, why not add it?  It helps reach 8 examples. Unless you're thinking of a stdin example as with `grep`...

Reply:
Ah, I see. In that case, it can perhaps stay here.

I would suggest though that parsing HTML etc with `grep`/`awk` is not something we should encourage though.

Reply:
No, certainly not.  That's just what I most frequently use `-o` for 👍🏻